features/vat-replacement/run.js

```diff
@@ -0,0 +1,213 @@
+    let errMsg  = null
+    if (invalidGermanTaxRateList.length>0) {
+        errMsg  = 'We are sorry, we would have to ask you to change the vat manually if applicable. ' +
+            'There seems to be a special case.' + '\n'
+        errMsg += await buildTaxRateDraftJson(invalidGermanTaxRateList, rateIdToCategoryMap)
+        console.error(errMsg)
+    } else if (standardVATs.length > 1) {
+        errMsg  = 'We are sorry but there are several tax rates for "DE" with a percentage ' +
+            'of ' + taxRate.standard.original*100 + ' percent. Please update the tax rate manually. ' + '\n'
+        errMsg += await buildTaxRateDraftJson(standardVATs, rateIdToCategoryMap)
+        console.error(errMsg)
+    } else if (reducedVATs.length > 1) {
+        errMsg  = 'We are sorry but there are several tax rates for "DE" with a percentage ' +
+            'of ' + taxRate.reduced.original*100 + ' percent. Please update the tax rate manually. ' + '\n'
+        errMsg += await buildTaxRateDraftJson(reducedVATs, rateIdToCategoryMap)
+        console.error(errMsg)
+    }
+
+    return validGermanTaxRateList
+}
+
+async function buildTaxRateDraftJson(taxRateList, rateIdToCategoryMap) {
+    let msg = ''
+    taxRateList.forEach(item => {
+        msg += 'Tax Category Name : '+rateIdToCategoryMap.get(item.id).name + '\n'
+        msg += JSON.stringify(item) + '\n'
+    })
+    return msg
+}
+
+async function buildRateIdToCategoryMap(items) {
+    const map = new Map()
+    for (const item of items) {
+        let categoryItem = {
+            id : item.id,
+            version : item.version,
+            name : item.name,
+            key : item.key
+        }
+        item.rates.forEach( rate => map.set(rate.id,  categoryItem))
+    }
+    return map
+}
+
+async function printUpdateJson(taxRateDraft, rateIdToCategoryMap ) {
+    const taxCategoryItem = rateIdToCategoryMap.get(taxRateDraft.id)
+
+    const updateJsonObj = {
+        version : taxCategoryItem.version,
+        actions : [
+            {
+                action: "replaceTaxRate",
+                taxRateId: taxRateDraft.id,
+                taxRate: {
+                    name: taxRateDraft.name,
+                    amount: taxRateDraft.amount,
+                    includedInPrice : taxRateDraft.includedInPrice,
+                    country: taxRateDraft.country
+                }
+            }
+        ]
+    }
+    console.log(JSON.stringify(updateJsonObj))
+
+}
+
+async function processTaxRate(ctpClient, validGermanTaxRateList, rateIdToCategoryMap, taxRateType, isDryRun) {
+
+    let taxRateDraftList =  validGermanTaxRateList.filter(rate => rate.amount == taxRateType.original)
+    if (taxRateDraftList.length>=1) {
+        console.log('Current tax rate would be replaced as below : ')
+        for (const taxRateDraft of taxRateDraftList) {
+            taxRateDraft.amount  = taxRateType.new
```

Comment:
It's not good practice to modify an element in an array, especially if you're going to reuse the array later. I guess you don't have a problem here as the next use will select different elements. But still I recommend to clone the object like this: `JSON.parse(JSON.stringify(food))`

Reply:
https://github.com/commercetools/commercetools-cli-scripts/pull/1/commits/f38cdb07ffcb18bc2be651ea14da96ddd7eb505e